pyproject.toml

```diff
@@ -1,2 +1,8 @@
+[project]
+requires-python = ">=3.6"
+
 [build-system]
-requires = ["setuptools", "wheel", "cython", "numpy"]
+requires = ["setuptools>=42",
+            "wheel",
+            "Cython",
+            "oldest-supported-numpy"]
```

Comment:
I'm unclear on whether `oldest-supported-numpy` is appropriate here. What happens if you disable PEP517?

Reply:
Builds that disable PEP517 use `setup.py` is used directly and skip `pyproject.toml`. GSD's `setup.py` imports numpy, so this style build requires that users manually install numpy first.

The `oldest-supported-numpy` [README](https://github.com/scipy/oldest-supported-numpy) answers some FAQs, like "Can I use this if my package requires a recent version of Numpy?" It points out that `pyproject.toml` describes the build system itself, while `setup.py` defines the installation requirements. While a project might use newer numpy python API calls, those may not be needed to build against the numpy C API.